std/assembly/string.ts

```diff
@@ -342,7 +345,122 @@ import { idof } from "./builtins";
     return changetype<String>(out); // retains
   }
 
+  replace(search: String, replacement: String): String {
+    assert(this !== null && search !== null && replacement !== null);
+    var len = this.length;
+    var slen = search.length;
+    if (len <= slen) {
+      return len < slen ? this : select<String>(replacement, this, search == this);
+    }
+    var index = this.indexOf(search);
+    if (~index) {
+      let rlen = replacement.length;
+      len = len - slen;
+      let olen = len + rlen;
+      if (olen) {
+        let out = __alloc(olen << 1, idof<String>());
+        memory.copy(out, changetype<usize>(this), <usize>index << 1);
+        memory.copy(
+          out + (<usize>index << 1),
+          changetype<usize>(replacement),
+          <usize>rlen << 1
+        );
+        memory.copy(
+          out + (<usize>(rlen + index) << 1),
+          changetype<usize>(this) + ((index + slen) << 1),
```

Comment:
Quite some `<usize>` conversion going on here, with one missing at this line. Perhaps casting the lengths to `isize` early can help to reduce conversion overhead?

Reply:
I remove some `usize` conversions

Reply:
What I meant in the former comment is that there are, besides others, multiple `<usize>index` in subsequent calls to memory.copy, resulting in redundant `i64.extend`s in 64-bit. Of course WASM64 isn't a thing yet (or anytime soon), but ideally stdlib would try to avoid redundancy here, for example by using `isize` as the var type as that can implicitly convert to usize without leading to an actual instruction. Ofc it's ok-ish to merge this without any 64-bit optimizations, but that might eventually backfire if we have to rethink all these things later.

Reply:
Do you mean just use `var index: isize = this.indexOf(search);` and skip all `<usize>index << 1` unnecessary conversion in this case?

Reply:
Something like that yeah,, essentially extending to i64 only once. Not all stdlib functions do this yet, but preparing what we can today will help later :)

Reply:
so I refactored in this style already in last commit

Reply:
Oh, must have missed. Great!